src/sentry/snuba/metrics/datasource.py

```diff
@@ -153,17 +164,53 @@ def get_metrics(projects: Sequence[Project]) -> Sequence[MetricMeta]:
 
     for derived_metric_mri in found_derived_metrics:
         derived_metric_obj = public_derived_metrics[derived_metric_mri]
-        metrics_meta.append(
-            MetricMeta(
-                name=get_public_name_from_mri(derived_metric_obj.metric_mri),
-                type=derived_metric_obj.result_type,
-                operations=derived_metric_obj.generate_available_operations(),
-                unit=derived_metric_obj.unit,
+        public_name = get_public_name_from_mri(derived_metric_obj.metric_mri)
+        if public_name is not None:
+            metrics_meta.append(
+                MetricMeta(
+                    name=public_name,
+                    type=derived_metric_obj.result_type,
+                    operations=derived_metric_obj.generate_available_operations(),
+                    unit=derived_metric_obj.unit,
+                )
             )
-        )
     return sorted(metrics_meta, key=itemgetter("name"))
 
 
+def get_custom_measurements(
+    projects: Sequence[Project],
+    organization: Optional[Organization] = None,
+    start: Optional[datetime] = None,
+    end: Optional[datetime] = None,
+) -> Sequence[MetricMeta]:
+    assert projects
+
+    metrics_meta = []
+    for metric_type in CUSTOM_MEASUREMENT_DATASETS:
+        for row in _get_metrics_for_entity(
+            entity_key=METRIC_TYPE_TO_ENTITY[metric_type],
+            projects=projects,
+            org_id=projects[0].organization_id if organization is None else organization.id,
+            start=start,
+            end=end,
+        ):
+            mri = reverse_resolve(row["metric_id"])
+            public_name = get_public_name_from_mri(mri)
+            # Custom measurements will get None from this function
```

Comment:
This statement is inaccurate. its custom measurements and privately ingested metric will return None according to the logic introduced in this PR